hibernate-core/src/main/java/org/hibernate/metamodel/mapping/internal/MappingModelCreationHelper.java

```diff
@@ -715,23 +744,36 @@ private static ForeignKeyDescriptor interpretKeyDescriptor(
 			assert bootValueMappingKey.getColumnSpan() == 1;
 			assert fkTarget instanceof BasicValuedModelPart;
 			final BasicValuedModelPart simpleFkTarget = (BasicValuedModelPart) fkTarget;
-
-			return new SimpleForeignKeyDescriptor(
-					( (AssociationType) bootValueMapping.getType() ).getForeignKeyDirection(),
-					getTableIdentifierExpression( bootValueMappingKey.getTable(), creationProcess ),
-					bootValueMappingKey.getColumnIterator().next().getText( dialect ),
-					simpleFkTarget.getContainingTableExpression(),
-					simpleFkTarget.getMappedColumnExpression(),
-					(BasicType) keyType
+			attributeMapping.setForeignKeyDescriptor(
+					new SimpleForeignKeyDescriptor(
+							getTableIdentifierExpression( bootValueMappingKey.getTable(), creationProcess ),
+							bootValueMappingKey.getColumnIterator().next().getText( dialect ),
+							simpleFkTarget.getContainingTableExpression(),
+							simpleFkTarget.getMappedColumnExpression(),
+							(BasicType) keyType
+					)
 			);
 		}
+		else if ( fkTarget instanceof EmbeddableValuedModelPart ) {
+			final EmbeddedForeignKeyDescriptor embeddedForeignKeyDescriptor =
+					buildEmbeddedForeignKeyDescriptor(
+							(EmbeddableValuedModelPart) fkTarget,
+							attributeMapping,
+							bootValueMapping,
+							dialect,
+							creationProcess
+					);
 
-		throw new NotYetImplementedFor6Exception(
-				"Support for composite foreign-keys not yet implemented: " + bootValueMapping.getRole()
-		);
+			attributeMapping.setForeignKeyDescriptor( embeddedForeignKeyDescriptor );
+		}
+		else {
+			throw new NotYetImplementedFor6Exception(
+					"Support for composite foreign-keys not yet implemented: " + bootValueMapping.getRole()
```

Comment:
I saw you used `foreign keys` elsewhere in this PR. Seems `foreign keys` is good enough without `-`. Again, super minor.